apps/studio/src/pages/sites/[siteId]/folders/[folderId]/index.tsx

```diff
@@ -109,23 +143,205 @@ const FolderPage: NextPageWithLayout = () => {
           </HStack>
         </VStack>
         <Box width="100%">
-          <ResourceTable siteId={siteId} resourceId={folderId} />
+          <ResourceTable
+            siteId={parseInt(siteId)}
+            resourceId={parseInt(folderId)}
+          />
         </Box>
       </VStack>
       <CreatePageModal
         isOpen={isPageCreateModalOpen}
         onClose={onPageCreateModalClose}
-        siteId={siteId}
+        siteId={parseInt(siteId)}
       />
       <CreateFolderModal
         isOpen={isFolderCreateModalOpen}
         onClose={onFolderCreateModalClose}
+        siteId={parseInt(siteId)}
+      />
+      <FolderSettingsModal
+        isOpen={isFolderSettingsModalOpen}
+        onClose={onFolderSettingsModalClose}
         siteId={siteId}
-        key={uniqueId()}
+        resourceId={parseInt(folderId)}
       />
     </>
   )
 }
 
 FolderPage.getLayout = AdminCmsSidebarLayout
 export default FolderPage
+
+interface FolderSettingsModalProps {
```

Comment:
should we extract all these non-page related components into their own `feature` directory instead of inlining in this page? We should keep pages scoped to rendering the page.

Reply:
ps i forgot about this - will extract into separate component file